Subject: [PATCH] atyfb: force 29MHz xtal on G3 PowerBooks

The atyfb does not work on my 233MHz PowerBook with Mach64 LP, when the
kernel is booted from firmware. aty_ld_pll_ct() returns 0x22 and xtal
remains at 14.31818. When booted from MacOS, aty_ld_pll_ct() returns 0x3c
and xtal is changed to 29.498928.
Google indicates that all 4 PowerBook models need the higher value.


---
 drivers/video/aty/atyfb_base.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/video/aty/atyfb_base.c	2007-07-08 19:32:17.000000000 -0400
+++ b/drivers/video/aty/atyfb_base.c	2007-08-27 14:01:19.000000000 -0400
@@ -2411,7 +2411,7 @@ static int __devinit aty_init(struct fb_
 				diff1 = -diff1;
 			if (diff2 < 0)
 				diff2 = -diff2;
-			if (diff2 < diff1) {
+			if (diff2 < diff1 || (M64_HAS(G3_PB_1024x768))) {
 				par->ref_clk_per = 1000000000000ULL / 29498928;
 				xtal = "29.498928";
 			}
